features/admin.roles.v2/api/roles.ts

```diff
@@ -186,6 +295,41 @@ export const useGetRoleById = <Data = RolesInterface, Error = RequestErrorInterf
     };
 };
 
+/**
+ * Retrieve Role details for a given role id using SCIM2 Roles V3 API.
+ *
+ * @param roleId - role id to retrieve role details
+ */
+export const useGetRoleByIdV3 = <Data = RolesInterface, Error = RequestErrorInterface>(
+    roleId: string
+): RequestResultInterface<Data, Error> => {
+    const requestConfig: RequestConfigInterface = {
+        headers: {
+            Accept: "application/json",
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: `${store.getState().config.endpoints.rolesV3}/${roleId}`
+    };
+
+    const {
+        data,
+        error,
+        isValidating,
+        mutate,
+        response
+    } = useRequest<Data, Error>(roleId ? requestConfig : null);
+
+    return {
+        data,
+        error,
+        isLoading: !error && !data,
```

Comment:
useRequest hook returns isLoading value too. Let's return it here.

Reply:
Let's create separate files for these custom hooks.

Reply:
Fixed with; https://github.com/wso2/identity-apps/pull/8184/commits/c0a4a2285a151f6204f093f4caf909c74f4c5233